File: tests/Foo/UnitTest1.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace Foo
{    
    [TestClass]
    public class UnitTest1
    {
        [TestMethod]
        public void TestMethod1()
        {
            Assert.IsTrue(true);
        }

        [TestMethod]
        public void TestMethod2()
        {
            Assert.IsTrue(false);
        }
    }
}

File: libraries/AdaptiveExpressionsCpp/AdaptiveExpressionsCpp/UnitTest1/UnitTest1.cpp
#include "pch.h"
#include "CppUnitTest.h"

#include "../Code/pch.h"
#include "ExpressionParser.h"
#include "FunctionUtils.h"

using namespace Microsoft::VisualStudio::CppUnitTestFramework;

namespace UnitTest1
{
	TEST_CLASS(UnitTest1)
	{
	public:
		
		TEST_METHOD(TestMethod1)
		{
            auto parsed = Expression::Parse("1 + 2");

            ValueErrorTuple valueAndError = parsed->TryEvaluate(nullptr);

            bool cast{};
            Assert::AreEqual(3, FunctionUtils::castToType<int>(valueAndError.first, cast));

            std::cout << "Error " << valueAndError.second << std::endl;
		}
	};
}
